src/test/java/Tests.java

```diff
@@ -0,0 +1,179 @@
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class Tests
```

Comment:
Tests look good to me, covers enough of the logic for now, but could add more in the future.